src/app/portlets/content-types/edit/content-types-edit.component.spec.ts

```diff
@@ -39,85 +46,167 @@ class TestContentTypeLayoutComponent {
     beforeEach(
         async(() => {
-            DOTTestBed.configureTestingModule({
-                declarations: [
-                    ContentTypesEditComponent,
-                    TestContentTypeFieldsDropZoneComponent,
-                    TestContentTypesFormComponent,
-                    TestContentTypeLayoutComponent
-                ],
-                imports: [
-                    RouterTestingModule.withRoutes([
-                        {
-                            component: ContentTypesEditComponent,
-                            path: 'test'
-                        }
-                    ])
-                ],
-                providers: [
-                    { provide: LoginService, useClass: LoginServiceMock },
-                    {
-                        provide: ActivatedRoute,
-                        useValue: { params: Observable.from([{ id: '1234' }]) }
-                    },
-                    CrudService,
-                    FieldService,
-                    Location
-                ]
-            });
+            DOTTestBed.configureTestingModule(configCreateMode);
 
             fixture = DOTTestBed.createComponent(ContentTypesEditComponent);
             comp = fixture.componentInstance;
             de = fixture.debugElement;
             el = de.nativeElement;
-            route = fixture.debugElement.injector.get(ActivatedRoute);
 
             crudService = fixture.debugElement.injector.get(CrudService);
             location = fixture.debugElement.injector.get(Location);
+            dotRouterService = fixture.debugElement.injector.get(DotRouterService);
+
+            fixture.detectChanges();
         })
     );
 
-    it('should have Content Types Layout', () => {
+    it('should has dialog opened by default', () => {
+        const dialog = de.query(By.css('p-dialog'));
+        expect(dialog).not.toBeNull();
+        expect(dialog.componentInstance.visible).toBeTruthy();
+    });
+
+    it('should have cancel button', () => {
+        const cancelButton = de.query(By.css('.content-type__cancel'));
+        expect(cancelButton === null).toBe(false);
+    });
+
+    it('should have save button and disabled by default', () => {
+        const saveButton = de.query(By.css('.content-type__save'));
+        expect(saveButton === null).toBe(false);
+        expect(saveButton.nativeElement.disabled).toBe(true);
+    });
+
+    it('should submit form when save button is clicked', () => {
+        const form: ContentTypesFormComponent = de.query(By.css('dot-content-types-form')).componentInstance;
+        spyOn(form, 'submitForm');
+        form.canSave = true;
+        fixture.detectChanges();
+
+        const saveButton = de.query(By.css('.content-type__save'));
+        saveButton.nativeElement.click();
+        expect(form.submitForm).toHaveBeenCalledTimes(1);
+    });
+
+    it('should bind save button disabled attribute to canSave property from the form', () => {
```

Comment:
we dont should test it, we should test that when the form change the save button is enabled end disabled in otherwise

Reply:
When the form change it change in the the content-type-form and that's the one in charge of set the canSave true or false and I use that property for the save button here in this component, I just need to make sure that the property is binded to the attr and I need to trust that the content-type-form is sending me the right value.